doc/dftb+/manual/dftbp.tex

```diff
@@ -3715,6 +3718,153 @@ \subsection{Casida}
+and ensemble DFT calculation is not compatible with time-dependent DFTB calculation.
+DFTB/SSR is not compatible with spin-polarisation, but it requires spin constants.
+
+In general, REKS calculation can be classified as single-state REKS, SA-REKS and SI-SA-REKS.
+In single-state REKS, only ground state is calculated and it can treat the state with multireference
+character. SA-REKS and SI-SA-REKS can calculate the vertical excitation energies. The difference
+is that the state-interaction term is considered in SI-SA-REKS so that more accurate states can
+be generated. The corresponding oscillator strengths as well as excited state geometry optimisation
+can be performed with these options, details of the resulting output files are given in appendix
+\ref{sec:reks_files}. DFTB/SSR is currently implemented for the LC-DFTB level of theory with external
+point charges or dispersion corrections. Third order SCC calculation can be considered in only
+energy evaluation. The gamma point periodic systems can be calculated using DFTB/SSR without
+range-separated functionals. Finally, only single-state REKS can calculate the stress as well as
+lattice optimization. The specifications for this block have the following properties:
+
+  \begin{ptable}
+    \kw{EnergyFunctional}                   & i & & 2 & \\
+    \kw{EnergyLevel}                          & i & & 1 & \\
+    \kw{UseSSRstate}                         & i & EnergyFunctional $\neq$ 1 & 1 & \\
+    \kw{TargetState}                           & i & & 1 & \\
+    \kw{TargetStateL}                          & i & EnergyFunctional $\neq$ 1, UseSSRstate = 0 & 0 & \\
+    \kw{InitialGuess}                           & i & & 1 & \\
+    \kw{FONmaxIter}                           & i & & 20 & \\
+    \kw{Shift}                                     & r & & 0.3 & \\
+    \kw{SpinTuning}                           & $(\text{r})^*$ & & \{\}~ & \\
+    \kw{TransitionDipole}                    & l & EnergyFunctional $\neq$ 1, TargetStateL = 0 & No & \\
+    \kw{GradientLevel}                        & i & & 1 & \\
+    \kw{CGmaxIter}                            & i & & 20 & \\
+    \kw{GradientTolerance}                 & r & & 1e-8 & \\
+    \kw{RelaxedDensity}                     & l & & No & \\
+    \kw{NonAdiabaticCoupling}            & l & EnergyFunctional $\neq$ 1, UseSSRstate = 1 & No & \\
+    \kw{PrintLevel}                             & i & & 1 & \\
+    \kw{MemoryLevel}                        & i & & 2 & \\
+  \end{ptable}
+
+  \begin{description}
+
+  \item[\is{EnergyFunctional}] Specifies the minimized energy functional in DFTB/SSR.
+    The minimized energy functionals can be PPS (\is{EnergyFunctional} = 1) or (PPS+OSS)/2
+    (\is{EnergyFunctional} = 2) state in DFTB/SSR(2,2) case. The former considers only
+    single-state REKS and the latter considers SA-REKS or SI-SA-REKS. The inclusion of
+    state-interaction term is determined by \is{UseSSRstate} option.
```

Comment:
@aradi For DFTB/SSR(2,2), there is only two choices for the `EnergyFunctional`. When the size of active space increases, the number of choice (and state) for the `EnergyFunctional` increases.
For example, the number of states used in SSR(4,4) is nine, so the length for the `EnergyFunctional` is too long if we includes all states. It may be `(PPS+OSS1+OSS2+ ... + DSPS)`. I think this is very complicated so I just use only number as choice. Is this explanation enough?

Reply:
@aradi @bhourahine For the naming of REKS input, I have one idea for this. General form is followed below.

`Energy = {
  Functional = {PPS OSS}
  Level = Same
}`

Here, `Functional` has the elements that we want to include into averaged functional in SA-REKS (This is same as `EnergyFunctional`). `Functional` can have following elements (`{PPS}`, `{PPS OSS}` or `{All}`) for SSR(2,2) case. The possible choice can be extended for SSR(4,4) case (`{PPS}`, `{PPS OSS1 OSS2}`, `{PPS DSPS}`, '{All}'). Thus, `Functional` determines which functional is included in the minimized functional in SA-REKS.

After that, we can decide the states to be included in SI-SA-REKS using `Level` variable. Here, there is two choices. One is same state used in SA-REKS and the other is all states. The role of variable `Level` is same as `EnergyLevel`. `Level` reads string (`Same` or `All`), not `1` or `2`.

For this case, I am not sure whether the parser routine can read only string as variable.
In addition, the variables such as `TargetState` related to energy calculation can be added into this `Energy` block.

I hope this explanation is clear to yours.